### PR TITLE
Changes for provisioning xamarin and UWP apps

### PR DESCRIPTION
**Pull request by raghusarangapani**
#### Fixes #.

- [x] Ran `flake8`
- [x] Ran `run_repo_tests.sh`

#### Changes proposed in this pull request:
- Changes for provisioning xamarin-ios and xamarin-android apps

### CODE REVIEW

File: keywords/TestServerAndroid.py [hunk 1]
```diff
@@ -13,20 +13,31 @@
...
+                self.download_source = "couchbase-lite-android-ee"
+            if debug_mode:
+                self.apk_name = "{}-debug.apk".format(apk_name_prefix)
+            else:
+                self.apk_name = "{}-release.apk".format(apk_name_prefix)
+            self.package_name = self.apk_name
+            self.device_enabled = False
+            self.installed_package_name = "com.couchbase.TestServerApp"
+            self.activity_name = self.installed_package_name + "/com.couchbase.CouchbaseLiteServ.MainActivity"
         else:
```
Comment (sridevi-15) [inline keywords/TestServerAndroid.py]: is this else for xamarin Android?

File: keywords/TestServerAndroid.py [hunk 2]
```diff
@@ -155,9 +170,10 @@ def start(self, logfile_name):
         self.logfile = open(logfile_name, "w+")
         self.process = subprocess.Popen(args=["adb", "logcat"], stdout=self.logfile)
 
-        activity_name = "com.couchbase.TestServerApp/com.couchbase.CouchbaseLiteServ.MainActivity"
+        # TestServer.Android/md57aea67c4d08319974f101b0b09ff509e.MainActivity
```
Comment (sridevi-15) [inline keywords/TestServerAndroid.py]: you can remove this?

File: keywords/TestServeriOS.py [hunk 3]
```diff
@@ -28,16 +27,29 @@ def __init__(self, version_build, host, port, community_enabled=None, debug_mode
...
+                self.app_dir = "CBLTestServer-iOS-community-{}".format(version_build)
+                self.package_name = "CBLTestServer-iOS-community-{}.zip".format(version_build)
+                self.app = "CBLTestServer-iOS"
+            else:
+                self.app_dir = "CBLTestServer-iOS-enterprise-{}".format(version_build)
+                self.package_name = "CBLTestServer-iOS-enterprise-{}.zip".format(version_build)
+                self.app = "CBLTestServer-iOS-EE"
+
+            self.bundle_id = "com.couchbase.CBLTestServer-iOS"
         else:
```
Comment (sridevi-15) [inline keywords/TestServeriOS.py]: not covered for community edition and Enterprise edition condition? I think it's ok as we are testing only enterprise edition here

File: keywords/TestServeriOS.py [hunk 4]
```diff
@@ -28,16 +27,29 @@ def __init__(self, version_build, host, port, community_enabled=None, debug_mode
...
-            self.app_dir = "CBLTestServer-iOS-enterprise-{}".format(version_build)
-            self.package_name = "CBLTestServer-iOS-enterprise-{}.zip".format(version_build)
-            self.app = "CBLTestServer-iOS-EE"
+            # Xamarin-ios
+            self.app_dir = "TestServer.iOS"
+            self.package_name = "TestServer.iOS.zip"
+            self.app = "TestServer.iOS"
+            self.bundle_id = "com.couchbase.TestServer-iOS"
```
Comment (sridevi-15) [inline keywords/TestServeriOS.py]: self.bundle_id is same for both if and else condition, you may want to put it outside of if else?

File: keywords/TestServerAndroid.py (hunk 1 above)
Comment (raghusarangapani) [reply keywords/TestServerAndroid.py]: Yes, I put a comment as well

File: keywords/TestServerAndroid.py (hunk 2 above)
Comment (raghusarangapani) [reply keywords/TestServerAndroid.py]: Done

File: keywords/TestServeriOS.py (hunk 3 above)
Comment (raghusarangapani) [reply keywords/TestServeriOS.py]: There is no CE app. Jim said that we will support CE version but will not carry out the same level of testing, so no need for a CE version of the app

File: keywords/TestServeriOS.py (hunk 4 above)
Comment (raghusarangapani) [reply keywords/TestServeriOS.py]: They are not the same, CBLTestServer-iOS vs TestServer-iOS

File: keywords/TestServerNetMsft.py [hunk 5]
```diff
@@ -45,54 +46,43 @@ def __init__(self, version_build, host, port, community_enabled=None):
...
         2. Extracts the package and removes the zip
         """
         if version_build is not None:
             self.version_build = version_build
-        version, build = version_and_build(self.version_build)
-        # download_url = "{}/couchbase-lite-net/{}/{}/TestServer.zip".format(LATEST_BUILDS, version, build)
-        # TODO: this is only for testing, remove download_url below line and uncomment later after testing don
-        download_url = "{}/couchbase-lite-net/{}/{}/TestServer.NetCore.zip".format(LATEST_BUILDS, version, build)
-        package_name = "TestServer.NetCore.zip"
-        build_name = "TestServer-Net-{}".format(self.version_build)
```
Comment (sridevi-15) [inline keywords/TestServerNetMsft.py]: need this for net-msft

Comment (raghusarangapani) [reply keywords/TestServerNetMsft.py]: Done

**Pull request merged**